src/protocol/attribution/aggregate_credit.rs

```diff
@@ -186,23 +160,40 @@ pub async fn aggregate_credit<F: Field>(
     Ok(result)
 }
 
-fn add_aggregation_bits_and_breakdown_keys<F: Field>(
+async fn add_aggregation_bits_and_breakdown_keys<F: Field, BK: BitArray>(
     ctx: &SemiHonestContext<'_, F>,
-    capped_credits: &[CreditCappingOutputRow<F>],
+    capped_credits: &[MCAggregateCreditInputRow<F>],
     max_breakdown_key: u128,
-) -> Vec<CappedCreditsWithAggregationBit<F>> {
+    num_multi_bits: u32,
+) -> Result<Vec<MCCappedCreditsWithAggregationBit<F>>, Error> {
     let zero = Replicated::ZERO;
     let one = ctx.share_of_one();
 
+    // Create secret shares of all possible breakdown key values
+    let unique_breakdown_keys = (0..max_breakdown_key)
+        .map(|i| XorReplicated::from_scalar(ctx.role(), BK::truncate_from(i)))
+        .collect::<Vec<_>>();
+
+    let converted_breakdown_keys = convert_all_bits(
+        &ctx.narrow(&Step::ModulusConversionForBreakdownKey),
+        &convert_all_bits_local(ctx.role(), &unique_breakdown_keys),
+        BK::BITS,
+        num_multi_bits,
+    )
+    .await
+    .unwrap();
```

Comment:
This seems wasteful. I don't think it's necessary to make this an async function, and actually perform real multiplications here given that we are just generating secret-sharings of known values.

I'm OK with removing this in a follow-up PR though.